src/cargo/ops/cargo_add/mod.rs

```diff
@@ -331,10 +336,25 @@ fn resolve_dependency(
         };
         selected
     } else if let Some(raw_path) = &arg.path {
-        let path = paths::normalize_path(&std::env::current_dir()?.join(raw_path));
-        let src = PathSource::new(&path);
+        let (path, base_name_and_value) = if let Some(base_name) = &arg.base {
+            let workspace_root = || Ok(ws.root_manifest().parent().unwrap());
+            let base_value = lookup_path_base(
+                &PathBaseName::new(base_name.clone())?,
+                gctx,
+                &workspace_root,
+                None,
+            )?;
+            (
+                base_value.join(raw_path),
```

Comment:
This assumes users will be doing
```console
$ cargo add --base foo --path bar
```
to get
```toml
bar = { base = "foo", path = "bar" }
```

Would users instead be expecting
```console
$ cargo add --base foo --path ../../../foo/bar
```
to get
```toml
bar = { base = "foo", path = "bar" }
```
or maybe even auto-base detection?

Reply:
I like the idea of `--path` always being relative the working directory, it also means that absolute paths will work as well.

I'm not a fan of auto-detection: I'd rather that the user be explicit.